prusti-viper/src/encoder/mir/specifications/interface.rs

```diff
@@ -0,0 +1,120 @@
+use crate::encoder::mir::specifications::specs::Specifications;
+use log::trace;
+use prusti_interface::specs::{typed, typed::DefSpecificationMap};
+use rustc_hash::FxHashMap;
+use rustc_hir::def_id::{DefId, LocalDefId};
+use std::cell::RefCell;
+
+pub(crate) struct SpecificationsState {
+    specs: RefCell<Specifications>,
+    pure_cache: RefCell<FxHashMap<DefId, bool>>,
+    trusted_cache: RefCell<FxHashMap<DefId, bool>>,
```

Comment:
Why is there a separate "cache" for these? IIUC to know if a method is pure or trusted, we need to first figure out its specification set anyway. That is, the `SpecMap` should always contain an entry for the `DefId`, so we might as well look up pureness or trustedness in that map, right?

Reply:
Caches removed